### PR TITLE
Feature/tqdm updater

### PR DESCRIPTION
**Pull request by CFlaniganMide**
Adds a tqdm based updater, I may add jupyter notebook support before actually merging.

### CODE REVIEW

File: examples/read_ide_file.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 # Loading an ide file is very simple, the supporting files are defined by
 # default in idelib.  This function returns a Document object which contains
 # the data for the given file
-doc = idelib.importFile('../test.ide')
+doc = idelib.importFile('../test.ide', updater=idelib.importer.TQDMUpdater())
```
Comment (StokesMIDE) [inline examples/read_ide_file.py:37]: The comment should probably call out and identify the updater

Comment (CFlaniganMide) [reply examples/read_ide_file.py:38]: agreed and fixed.

**Review by StokesMIDE — APPROVED** [review]
I'm good with it. Should anyone else weigh in?

File: examples/read_ide_file.py (hunk 1 above)
Comment (CFlaniganMide) [reply examples/read_ide_file.py:38]: @CrepeGoat @SamRagusa either of you guys want to test this in a jupyter notebook/colab to see if it looks good?

**Comment by CFlaniganMide** [comment]
> 
> 
> I'm good with it. Should anyone else weigh in?

@CrepeGoat @SamRagusa Either of you guys want to try this in a jupyter notebook/colab to check it out?

**Review by CrepeGoat — APPROVED** [review]
Just ran on Colab, looks good!
https://colab.research.google.com/drive/1OhHFFqBbCRE3RBGztYismMdW1H5EFL6f?usp=sharing

**Pull request merged**